spec/predicates/runtime-trace.md

```diff
@@ -0,0 +1,274 @@
+            ]
+        },
+        "metadata": {
+            "buildStartedOn": "<TIMESTAMP>",
+            "buildFinishedOn": "<TIMESTAMP>"
+        }
+    }
+Ex: GitLab Runner ID}
+```
+
+### Parsing Rules
+
+This predicate follows the in-toto attestation parsing rules. Summary:
+
+-   Consumers MUST ignore unrecognized fields.
+-   The `predicateType` URI includes the major version number and will always
+    change whenever there is a backwards incompatible change.
+-   Minor version changes are always backwards compatible and “monotonic.” Such
+    changes do not update the `predicateType`.
+-   Producers MAY add extension fields using field names that are unlikely to
+    collide with names used by other producers. Field names SHOULD avoid using
+    characters like `.` and `$`.
+-   Fields marked _optional_ MAY be unset or null, and should be treated
+    equivalently. Both are equivalent to empty for object or array values.
+
+### Fields
+
+`monitor` _object_, _required_
+
+Identifies the specific monitor instance used to trace the runtime.
+
+`monitor.id` _string (URI)_, _required_
+
+URI indicating the monitor’s identity.
+
+`monitor.configSource` ResourceDescriptor, _optional_
+
+Effectively a pointer to the monitor's configuration.
+
+`monitor.tracePolicy` _object_, _optional_
+
+Indicates the trace policy used for the monitoring event that generated the
+runtime trace. The trace policy's format is dependent on the monitor used, and
+so it must be parsed after identifying the monitor using `monitor.id`.
+
+FIXME: should this be optional?
+
+`monitoredProcess` _object_, _required_
+
+Identifies the process being monitored.
+
+`monitoredProcess.hostID` _string (URI)_, _required_
+
+URI indicating the process host’s identity.
+
+`monitoredProcess.type` _string (TypeURI)_, _required_
+
+URI indicating the type of process performed. Ex: when monitoring a build, this
+field records the build type.
+
+`monitoredProcess.event` _string_, _required_
+
+String identifying the specific job or task associated with the attestation.
+
+`monitorLog` _object_, _required_
+
+Record of events that were traced by the monitor. At least one of `process`,
+`network`, and `fileAccess` must be present.
+
+`monitorLog.process` _list of objects_, _optional_
+
+Record of processes observed by monitor. The exact format of this field is
+currently dependent on the monitor, and can be determined using `monitor.id`.
+In future, after consulting different monitors, this field may have a consistent
+schema.
+
+`monitorLog.network` _list of objects_, _optional_
+
+Record of network activity observed by monitor. The exact format of this field
+is currently dependent on the monitor, and can be determined using `monitor.id`.
+In future, after consulting different monitors, this field may have a consistent
+schema.
+
+`monitorLog.fileAccess` _list of objects_, _optional_
+
+Record of files accessed during the monitored process. A complete list of
```

Comment:
Is the idea that fileAccess is sufficiently well defined that all monitors can use it?  If so the various fields should probably be defined here?  Can also be TBD for v0.1

Reply:
Updated to be list of `ResourceDescriptor` objects.

Reply:
I primarily foresee `name` and `digest` being used so I think this should suffice until a use case indicates otherwise.

Reply:
That sounds good.  How would you feel about adding some text to that effect?

Reply:
I tweaked the text to explicitly call out ResourceDescriptor and the data that may be captured in an instance.